File: scaffold/src/main/java/top/xuqingquan/web/publics/AgentWebUIControllerImplBase.java
package top.xuqingquan.web.publics;

import android.app.Activity;
import android.os.Handler;

public class AgentWebUIControllerImplBase extends AbsAgentWebUIController {

    public static AbsAgentWebUIController build() {
        return new AgentWebUIControllerImplBase();
    }

    @Override
    public void onJsAlert(android.webkit.WebView view, String url, String message) {
        getDelegate().onJsAlert(view, url, message);
    }

    @Override
    public void onJsAlert(com.tencent.smtt.sdk.WebView view, String url, String message) {
        getDelegate().onJsAlert(view, url, message);
    }

    @Override
    public void onOpenPagePrompt(android.webkit.WebView view, String url, Handler.Callback callback) {
        getDelegate().onOpenPagePrompt(view, url, callback);
    }

    @Override
    public void onOpenPagePrompt(com.tencent.smtt.sdk.WebView view, String url, Handler.Callback callback) {
        getDelegate().onOpenPagePrompt(view, url, callback);
    }

    @Override
    public void onJsConfirm(android.webkit.WebView view, String url, String message, android.webkit.JsResult jsResult) {
        getDelegate().onJsConfirm(view, url, message, jsResult);
    }

    @Override
    public void onJsConfirm(com.tencent.smtt.sdk.WebView view, String url, String message, com.tencent.smtt.export.external.interfaces.JsResult jsResult) {
        getDelegate().onJsConfirm(view, url, message, jsResult);
    }

    @Override
    public void onSelectItemsPrompt(android.webkit.WebView view, String url, String[] ways, Handler.Callback callback) {
        getDelegate().onSelectItemsPrompt(view, url, ways, callback);
    }

    @Override
    public void onSelectItemsPrompt(com.tencent.smtt.sdk.WebView view, String url, String[] ways, Handler.Callback callback) {
        getDelegate().onSelectItemsPrompt(view, url, ways, callback);
    }

    @Override
    public void onForceDownloadAlert(String url, Handler.Callback callback) {
        getDelegate().onForceDownloadAlert(url, callback);
    }

    @Override
    public void onJsPrompt(android.webkit.WebView view, String url, String message, String defaultValue, android.webkit.JsPromptResult jsPromptResult) {
        getDelegate().onJsPrompt(view, url, message, defaultValue, jsPromptResult);
    }

    @Override
    public void onJsPrompt(com.tencent.smtt.sdk.WebView view, String url, String message, String defaultValue, com.tencent.smtt.export.external.interfaces.JsPromptResult jsPromptResult) {
        getDelegate().onJsPrompt(view, url, message, defaultValue, jsPromptResult);
    }

    @Override
    public void onMainFrameError(android.webkit.WebView view, int errorCode, String description, String failingUrl) {
        getDelegate().onMainFrameError(view, errorCode, description, failingUrl);
    }

    @Override
    public void onMainFrameError(com.tencent.smtt.sdk.WebView view, int errorCode, String description, String failingUrl) {
        getDelegate().onMainFrameError(view, errorCode, description, failingUrl);
    }

    @Override
    public void onShowMainFrame() {
        getDelegate().onShowMainFrame();
    }

    @Override
    public void onLoading(String msg) {
        getDelegate().onLoading(msg);
    }

    @Override
    public void onCancelLoading() {
        getDelegate().onCancelLoading();
    }


    @Override
    public void onShowMessage(String message, String from) {
        getDelegate().onShowMessage(message, from);
    }

    @Override
    public void onPermissionsDeny(String[] permissions, String permissionType, String action) {
        getDelegate().onPermissionsDeny(permissions, permissionType, action);
    }

    @Override
    public void bindSupportWebParent(WebParentLayout webParentLayout, Activity activity) {
        getDelegate().bindSupportWebParent(webParentLayout, activity);
    }


}

File: scaffold/src/main/java/top/xuqingquan/web/publics/AgentWebUIControllerImplBase.kt
package top.xuqingquan.web.publics

import android.app.Activity
import android.os.Handler
import android.webkit.JsPromptResult
import android.webkit.JsResult
import android.webkit.WebView
import com.tencent.smtt.sdk.WebView as X5WebView
import com.tencent.smtt.export.external.interfaces.JsResult as X5JsResult
import com.tencent.smtt.export.external.interfaces.JsPromptResult as X5JsPromptResult

class AgentWebUIControllerImplBase : AbsAgentWebUIController() {

    override fun onJsAlert(view: WebView, url: String, message: String) {
        getDelegate().onJsAlert(view, url, message)
    }

    override fun onJsAlert(view: X5WebView, url: String, message: String) {
        getDelegate().onJsAlert(view, url, message)
    }

    override fun onOpenPagePrompt(view: WebView, url: String, callback: Handler.Callback) {
        getDelegate().onOpenPagePrompt(view, url, callback)
    }

    override fun onOpenPagePrompt(view: X5WebView, url: String, callback: Handler.Callback) {
        getDelegate().onOpenPagePrompt(view, url, callback)
    }

    override fun onJsConfirm(view: WebView, url: String, message: String, jsResult: JsResult) {
        getDelegate().onJsConfirm(view, url, message, jsResult)
    }

    override fun onJsConfirm(view: X5WebView, url: String, message: String, jsResult: X5JsResult) {
        getDelegate().onJsConfirm(view, url, message, jsResult)
    }

    override fun onSelectItemsPrompt(
        view: WebView, url: String, ways: Array<String>, callback: Handler.Callback
    ) {
        getDelegate().onSelectItemsPrompt(view, url, ways, callback)
    }

    override fun onSelectItemsPrompt(
        view: X5WebView, url: String, ways: Array<String>, callback: Handler.Callback
    ) {
        getDelegate().onSelectItemsPrompt(view, url, ways, callback)
    }

    override fun onForceDownloadAlert(callback: Handler.Callback) {
        getDelegate().onForceDownloadAlert(callback)
    }

    override fun onDownloadPrompt(fileName: String, callback: Handler.Callback) {
        getDelegate().onDownloadPrompt(fileName, callback)
    }

    override fun onJsPrompt(
        view: WebView, url: String, message: String,
        defaultValue: String, jsPromptResult: JsPromptResult
    ) {
        getDelegate().onJsPrompt(view, url, message, defaultValue, jsPromptResult)
    }

    override fun onJsPrompt(
        view: X5WebView, url: String, message: String,
        defaultValue: String, jsPromptResult: X5JsPromptResult
    ) {
        getDelegate().onJsPrompt(view, url, message, defaultValue, jsPromptResult)
    }

    override fun onMainFrameError(
        view: WebView, errorCode: Int, description: String, failingUrl: String
    ) {
        getDelegate().onMainFrameError(view, errorCode, description, failingUrl)
    }

    override fun onMainFrameError(
        view: X5WebView, errorCode: Int, description: String, failingUrl: String
    ) {
        getDelegate().onMainFrameError(view, errorCode, description, failingUrl)
    }

    override fun onShowMainFrame() {
        getDelegate().onShowMainFrame()
    }

    override fun onLoading(msg: String) {
        getDelegate().onLoading(msg)
    }

    override fun onCancelLoading() {
        getDelegate().onCancelLoading()
    }


    override fun onShowMessage(message: String, intent: String) {
        getDelegate().onShowMessage(message, intent)
    }

    override fun onPermissionsDeny(
        permissions: Array<String>, permissionType: String, action: String
    ) {
        getDelegate().onPermissionsDeny(permissions, permissionType, action)
    }

    override fun bindSupportWebParent(webParentLayout: WebParentLayout, activity: Activity) {
        getDelegate().bindSupportWebParent(webParentLayout, activity)
    }

    companion object {

        @JvmStatic
        fun build(): AbsAgentWebUIController {
            return AgentWebUIControllerImplBase()
        }
    }


}
